### ISSUE TITLE
Remove non-working/obsolete tests?

### ISSUE THREAD

**Issue by JJ** [perl6/roast#533]
While looking at #532 I've realized [this](https://github.com/perl6/roast/blob/master/S14-traits/variables.t) it's 8 years old, and that the last commit, itself, says it does not work in nom.
Wouldn't it be a good idea to simply remove tests that are not working? Or fix them? Isn't this spec supposed to be *the* spec?

**Comment by JJ** [comment]
Effectively, [that file (along with packages.t is not listed in spectest.data in Rakudo)](https://github.com/rakudo/rakudo/blob/8f1b1babd2c33ed4042969ecfc4a078307651252/t/spectest.data). But that's confusing, and it also means that variable traits are not tested. Is that correct?

**Comment by JJ** [comment]
Ping?
Here's my version of that test, before I obliterate it
```
use v6;
use Test;

plan 5;

# L<S14/Traits/>

my @var_names;
multi trait_mod:<is>(Variable $a, :$noted!) {
    push @var_names, $a.name;
}

role doc { has $.doc is rw }
# multi trait_mod:<is>($a, $arg, :$doc!) {
#     $a.container.VAR does doc.new(doc => $arg);
# }

multi trait_mod:<is>(Variable $a, :$docced!) {
    $a does doc.new(doc => $docced);
}


my $a is noted;
my %b is noted;
my @c is noted;

@var_names .= sort;
is +@var_names, 3, 'have correct number of names noted from trait applied by name';
is @var_names, ['$a','%b','@c'], 'trait recorded correct information';


my $dog is docced('barks');
my @birds is docced('tweet');
my %cows is docced('moooo');

say $dog.^name;
is $dog.VAR.doc, 'barks', 'trait applied to scalar variable correctly';
is @birds.doc,   'tweet', 'trait applied to array variable correctly';
is %cows.doc,    'moooo', 'trait applied to hash variable correctly';

# vim: ft=perl6
```

**Comment by Altai-man** [comment]
1)What can "obsolete" mean here? I mean, what's the criteria?
2)If we have tests which are broken, they have to be fixed, not removed. If we have tests which test NYI, we have to implement, not remove tests

**Comment by JJ** [comment]
I mainly mean tests that are not part of the spec, and thus they are not run. This is one of them.

**Comment by coke** [comment]
> If we have tests which test NYI, we have to implement, not remove tests

Not necessarily. Plans have changed in the last 20 years, something that was once intended to be implemented may not be on roadmap anymore.
